app/controllers/NotConnectedCheckYourAnswersController.scala

```diff
@@ -0,0 +1,151 @@
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+
+import java.time.Instant
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+
+@Singleton
+class NotConnectedCheckYourAnswersController @Inject()
+( repository: FormDocumentRepository,
+  submissionConnector: SubmissionConnector,
+  refNumAction: RefNumAction,
+  cache: MongoSessionRepository,
+  audit: Audit,
+  cc: MessagesControllerComponents,
+  notConnectedCheckYourAnswers: views.html.notConnectedCheckYourAnswers,
+  confirmNotConnectedView: views.html.confirmNotConnected,
+  errorView: views.html.error.error
+)(implicit ec: ExecutionContext)
+  extends FrontendController(cc) {
+
+  val logger = Logger(classOf[NotConnectedCheckYourAnswersController])
+
+  def findSummary(implicit request: RefNumRequest[_]) = {
+    repository.findById(SessionId(hc), request.refNum) flatMap {
+      case Some(doc) => Option(SummaryBuilder.build(doc))
+      case None => None
+    }
+  }
+
+  def findNotConnected(sum: Summary)(implicit hc: HeaderCarrier) = {
+    getNotConnectedFromCache.flatMap { notConnected =>
+      getPreviouslyConnectedFromCache().flatMap { previouslyConnected =>
+        Option(NotConnectedSummary(sum, previouslyConnected, notConnected))
+      }
+    }
+  }
+
+  def removeSession(implicit request: RefNumRequest[_]) = {
+    repository.remove(SessionId(hc))
+  }
+
+  def onPageView = refNumAction.async { implicit request =>
+    findSummary.flatMap { summary =>
+      findNotConnected(summary.get).map {
+        case Some(notConnectedSummary) => Ok(notConnectedCheckYourAnswers(notConnectedSummary))
+        case None => {
+          logger.error(s"Could not find document in current session - ${request.refNum} - ${hc.sessionId}")
+          InternalServerError(errorView(500))
+        }
+      }
+    }
+  }
+
+  def onPageSubmit = refNumAction.async { implicit request =>
+    findSummary.flatMap {
+      case Some(summary) => {
+        audit.sendExplicitAudit("NotConnectedSubmission", Json.obj(Audit.referenceNumber -> summary.referenceNumber))
+        submitToHod(summary).map { _ =>
+          Redirect(routes.NotConnectedCheckYourAnswersController.onConfirmationView)
+        }.recover {
+          case e: Exception => {
+            logger.error(s"Could not send data to HOD - ${request.refNum} - ${hc.sessionId}")
+            InternalServerError(errorView(500))
+          }
+        }
+      }
+    }
+  }
+
+  def onConfirmationView() = refNumAction.async { implicit request =>
+    val feedbackForm = Survey.completedFeedbackForm.bind(
+      Map("journey" -> NotConnectedJourney.name)
+    ).discardingErrors
+
+    findSummary.flatMap {
+      case Some(summary) => {
+        removeSession
+          .map(_ => Ok(confirmNotConnectedView(feedbackForm)))
+      } //.withNewSession
+      case None => {
+        logger.error(s"Could not find document in current session - ${request.refNum} - ${hc.sessionId}")
```

Comment:
Logging is not necesarry. This is completely legal state. For example, user switch language or reload page.

Reply:
Logging removed.